MetaMask/metamask-extension
# Part #19737 - Replace deprecated FormField component with TextField, Label, HelpText

## Explanation

This pull request addresses the Issue of replacing the deprecated `FormField` component with the modernized `Label`, `TextField` and `HelpText` components from our component library. Additionally, it includes updates to the constants previously used in the codebase, changing `DISPLAY` to `Display`, `FLEX_DIRECTION` to `FlexDirection`, and `BLOCK_SIZES` to `BlockSize`.

* Part #19737 

## Changes made
* Replaced instances of the deprecated `FormField` component with `TextField`, `Label` and `HelpText` components.
* Updated constants: `DISPLAY` to `Display`, `FLEX_DIRECTION` to `FlexDirection`, and `BLOCK_SIZES` to `BlockSize`.

## Screenshots/Screencaps
### Before
Some of the icons and text were unaligned and inconsistent. The input type is also text when it is a number input 

https://github.com/MetaMask/metamask-extension/assets/8112138/aef8a0d1-cbed-4a92-9431-d985020f1dad

Storybook also didn't work as expected you can't edit the input

https://github.com/MetaMask/metamask-extension/assets/8112138/289c3e96-48ac-484e-bad2-4ba91558ae39

### After
Icons are now consistent in the label and tooltip, the alignment of the max button is centered the label has also been updated to align with the design system


https://github.com/MetaMask/metamask-extension/assets/8112138/8a500fdb-e394-4463-82a6-98ed521db3d6


Storybook now works as expected being able to use the input to enter numbers

https://github.com/MetaMask/metamask-extension/assets/8112138/c120fd5c-6ce2-46ca-a515-ac11331dce4d


## Manual Testing Steps
Storybook
- Pull this branch
- Serach `CustomSpendingCap` in stroybook
- Check against `develop` storybook https://metamask.github.io/metamask-storybook/?path=/story/components-app-customspendingcap--custom-spending-cap-story&args=dappProposedValue:234234234

Check custom spending cap in app
- Go to test dapp https://metamask.github.io/test-dapp/
- Click `CREATE TOKEN` you will need test eth
- Wait for transaction
- Click `APPROVE TOKENS` 
- Use Custom Spending Cap component 

## Pre-merge author checklist


## Pre-merge reviewer checklist
